### DISCUSSION TITLE
How does your seach nodes look like?

### DISCUSSION THREAD

**Discussion by OzzyKampha** [Security-Onion-Solutions/securityonion#10872] [Unsupported Versions] [answered]
I was wondering if someone could share their storage sizing. How large are they in term of CPU RAM and storage. And when do you create a new one?

**Comment by vanh20** [comment]
Hey Ozzy, check [this](https://docs.securityonion.net/en/2.3/hardware.html#search-node) out.

**Reply by OzzyKampha** [reply]
I have seen it before, but i was wondering what people are using in their env, since 8 cores for a Elastic node is very low

**Reply by vanh20** [reply]
Let's hope there are generous takers. We would all benefit from their experience.

**Reply by vanh20** [reply]
Also, try to experiment on having it all run without using a swapfile. BTW, do you still use HDDs? I've tried it before on an SSD drive on a system with a 32GB memory (including the video ram) and it was fast. Production-wise, opt for the fastest hardware possible. So I guess, the next consideration is how much traffic is in your search nodes and how much you keep for x number of days. 

So when to scale up on anything? On your SOC, you have the Tools section and have the [Grafana Menu](https://docs.securityonion.net/en/2.3/grafana.html?highlight=grafana) to open your Grafana dashboard. You will see there how much of your CPU, RAM, DISK, NIC are utilized. So if they stay long on being fully utilized, then it's time to scale-up that device, or scale-horizontally on your search nodes as a whole. 

There's this [old discussion](https://github.com/Security-Onion-Solutions/securityonion/discussions/3162) I started. There were suggestions on how to optimize things. Maybe you'd find it useful.

**Comment by OzzyKampha** [comment, marked as answer]
I was able to find the ratio elastic is using for their [Cloud:](https://www.elastic.co/guide/en/cloud/current/ec-default-azure-configurations.html)

Hot 35:1 disk:mem 0.533 vcpu/RAM
Warm/cold 200:1 disk:mem 0.133

So in the scenario of 64 Gig Mem for 32 Gig heap:
- Hot size:64\*35=2,240GB≈2.2TB, CPU:0.533\*64=34.112≈35 cores ≈36 whole set cores 
- Warm/cold size:64\*200=12,800GB≈12.8TB, CPU: 0.133\*64=8.512≈9≈10

Each node should be then: 
Cpu:35-36, RAM:64, Disk:2.2TB+10% watermark

Cpu:9-10, RAM:64, Disk:12.8TB+10% watermark

So the question is if this apply to SO

----------------------------
Edit: since our retention_pct is default 50% then we need to rise the storage to the doble to meet these ratios.

So hot: 2.2TB-> 4.4TB
  Warm: 12.8TB -> 25,6TB

**Reply by vanh20** [reply]
It can. And one needs to utilize the [right](https://learn.microsoft.com/en-us/azure/virtual-machines/fsv2-series) hardware for it as per your example. Elastic also has recommendations on [when to scale](https://www.elastic.co/guide/en/cloud/current/ec-planning.html#ec-workloads) and uses some form of performance [metrics](https://www.elastic.co/guide/en/cloud/current/ec-saas-metrics-accessing.html#ec-saas-metrics-accessing) which is actually a bit more interesting than SO's Grafana as it has nice explanations that go with it. Now, you may want to compare it with what SO Actually recommends for [Azure](https://sec-on/docs/cloud-azure.html?highlight=azure#distributed-grid). I would be interested on seeing [this](https://learn.microsoft.com/en-us/azure/virtual-machines/dav4-dasv4-series) also. 

Overall, since upscaling our vCPU or RAM is easy on the cloud, doing some experiment using Grafana for perfomance tuning is easy. And then, there's the autoscaling feature of AWS if it is supported by your instance though I'm not so sure how it would play out.
